qt/aqt/browser/card_info.py

```diff
@@ -47,10 +64,87 @@ def _setup_ui(self, card_id: CardId) -> None:
         self.setLayout(layout)
 
         self.web.eval(
-            f"anki.cardInfo(document.getElementById('main'), {card_id}, true);"
+            "const cardInfo = anki.cardInfo(document.getElementById('main'));\n"
+            "cardInfo.then((c) => c.$set({ includeRevlog: true }));"
         )
+        self.update_card(card_id)
+
+    def update_card(self, card_id: CardId | None) -> None:
+        val = "undefined" if card_id is None else card_id
+        self.web.eval(f"cardInfo.then((c) => c.$set({{ cardId: {val} }}));")
```

Comment:
This is fine as is, but the approach I'd take is:

- default to null instead of undefined in the component, which I personally think slightly better represents "deliberately set to nothing" instead of "omitted"
- then instead of manually making string values, you can just use `cardId: {json.dumps(card_id)}` in the eval line, and it will convert the None to null.